utils/swift_build_support/swift_build_support/products/swiftdriver.py

```diff
@@ -83,8 +87,14 @@ def run_build_script_helper(action, host_target, product, args):
         install_destdir = swiftpm.SwiftPM.get_install_destdir(args,
                                                               host_target,
                                                               product.build_dir)
-    toolchain_path = targets.toolchain_path(install_destdir,
-                                            args.install_prefix)
+
+    if local_compiler_build:
+      # If building with the host toolchain for use with a local compiler build,
+      # use the toolchain which is supplying the `swiftc`.
+      toolchain_path = os.path.dirname(os.path.dirname(os.path.abspath(product.toolchain.swiftc)))
```

Comment:
It feels like there should be a better way to find this path, but I haven't been able to find one yet.